src/components/Map.vue

```diff
@@ -55,30 +65,60 @@ export default {
     },
   },
   computed: mapState({
+    /*
+    Add asset geometry to state
+    */
     mapboxAPIKey: state => state.shell.mapboxAPIKey,
   }),
   mounted() {
+    /*
+    Render asset geometry
+    */
     this.map = window.farmOS.map.create(this.id, this.options);
     if (this.geojson.url) {
       this.layers.geojson = this.map.addLayer('geojson', this.geojson);
     }
-    if (!this.drawing && this.wkt.wkt && this.wkt.wkt !== 'GEOMETRYCOLLECTION EMPTY') {
-      this.layers.wkt = this.map.addLayer('wkt', this.wkt);
-      this.map.zoomToLayer(this.layers.wkt);
-    }
-    if (this.geojson.url && (!this.wkt.wkt || this.wkt.wkt === 'GEOMETRYCOLLECTION EMPTY')) {
-      this.layers.geojson.getSource().once('change', () => { this.map.zoomToVectors(); });
-    }
-    if (this.drawing) {
-      if (this.wkt.wkt && this.wkt.wkt !== 'GEOMETRYCOLLECTION EMPTY') {
-        this.layers.wkt = this.map.addLayer('wkt', this.wkt);
-        this.map.addBehavior('edit', { layer: this.layers.wkt });
-        this.map.addBehavior('measure', { layer: this.layers.wkt });
-        this.map.zoomToLayer(this.layers.wkt);
-      } else {
-        this.map.addBehavior('edit');
-        this.map.addBehavior('measure', { layer: this.map.edit.layer });
+    let hasLayers = false;
+    this.wkt.forEach((wktElement) => {
+      /*
+      When a movement layer is present, zoom to the movement layer.
+      Otherwise, zoom to the previous layer.
+      */
+      if (!this.drawing
+        && wktElement.wkt
+        && wktElement.wkt !== 'GEOMETRYCOLLECTION EMPTY'
+        && wktElement.wkt !== null) {
+        this.layers[wktElement.title] = this.map.addLayer('wkt', wktElement);
+        if (hasLayers && this.layers.movement) {
+          this.map.zoomToLayer(this.layers.movement);
+        } else {
+          this.map.zoomToLayer(this.layers[wktElement.title]);
+          hasLayers = true;
+        }
+      }
+      if (this.drawing) {
+        if (wktElement.wkt
+          && wktElement.wkt !== 'GEOMETRYCOLLECTION EMPTY'
+          && wktElement.wkt !== null) {
+          if (wktElement.title === 'movement') {
```

Comment:
I'm a little wary that we're hardcoding movements into this component which should be a generic mapping component. Is there a way perhaps instead that we could allow different layers to be assigned some kind of zoom "weight"? Something like an optional number associated with each layer, and whichever layer has the highest number, we zoom to it?

Reply:
You're absolutely right - we want the app core to accommodate other map implementations, and hard-coding movements doesn't facilitate that.  Thank you for catching this!!

I'm 100% behind your suggestion of including a weight in each wkt object.  That's an elegant solution, and it sounds like it should work!

Reply:
Cool! Yea use your judgment insofar as what's best. An integer seems most intuitive, but then you still have to decide if they should be ranked in ascending or descending order. Another idea that occurs to me, which could be a little more complicated, but it might be possible to just have some sort of boolean value for all the layers we want within the zoom view, then generate some kind of temporary bounding box around any and all layers with a `true` value, and zoom to that. That might be something for farmOS-map itself (what say ye, @mstenta?), but it could be interesting to test these kinds of ideas out in Field Kit first. Then again, using ints as weights might be the most expedient thing at this time; we could circle back and evaluate the bounding box idea some other time.

Reply:
Hey @jgaehring, I've got this working!  I added a weight attribute to each element of the wkt array.  In this weighting system, the map zooms to the lowest-weighted layer that possesses a geometry.

I also found that each wkt element needs a canEdit attribute, since we only want to edit movement geometries on the movements map.

Reply:
Excellent! Looks great!

Oh good call with the canEdit attribute. I hadn't considered that, but yes, that makes perfect sense.